packages/tooltip/src/index.tsx

```diff
@@ -285,14 +290,74 @@ function useTooltip<T extends HTMLElement>({
     return () => ownerDocument.removeEventListener("keydown", listener);
   }, []);
 
-  function handleMouseEnter() {
+  React.useEffect(() => {
+    /*
+      This is a workaround for using tooltips with disabled controls in Safari.
+      Safari fires `pointerenter` but does not fire `pointerleave`
+      and `onPointerEventLeave` added to the trigger element will not work
+    */
+
+    if (!("PointerEvent" in window)) return;
+
+    let ownerDocument = getOwnerDocument(ownRef.current)!;
+
+    function listener(event: MouseEvent) {
+      // @ts-ignore `disabled` does not exist on HTMLDivElement but tooltips can be used with different elements
+      if (state !== VISIBLE || !ownRef.current?.disabled) return;
```

Comment:
Instead of checking the ref's element here, have you considered accepting `disabled` as an additional prop for the hook? That would allow us to use it as a dependency for the effect and avoid attaching useless event listeners for non-disabled components.

Reply:
Not sure if I understood correctly. Do you mean something like:
```tsx
<Tooltip id="wow" label="Notifications" isTriggerDisabled={isDisabled}>
  <Button isDisabled={isDisabled} />
</Tooltip>
```
?

Since this whole event listener is a workaround only for one browser I wouldn't add any additional prop to the hook or component. One more event listener in addition to another 4 doesn't make a huge impact and it's created only once. And it can be safely removed when/if the issue will be fixed in Safari.

Reply:
Yeah it's a bit awkward as for the component API, but I think it makes a little more sense for the hook API. It would compose pretty naturally on buttons since `disabled` is a built-in prop for JSX (less-so if you have your own button API with something like `isDisabled`, but still). I'd be comfortable leaving it out of the component for the time being and considering that API separately while supporting the core logic in the hook.

```js
const CustomTooltipButton = React.forwardRef(({ label, ...props }, ref) => {
  // as long as `disabled` is a prop, this should work without changes to the existing API
  let [trigger, tooltip] = useTooltip({ ...props, ref });
  return (
    <React.Fragment>
      <button {...props} {...trigger} />
      <CustomTooltipPopup label={label} {...tooltip} />
    </React.Fragment>
  );
});
```

Reply:
Actually, I just realized since we're already using `cloneElement` in `Tooltip` this is absolutely trivial to solve there as well. If `disabled` is passed to the child this should all just work with the addition of one line:

```js
disabled: child.props.disabled
```

Reply:
Pushed an update.

I'm a bit concerned about the meaning of `disabled` param here though. Knowing implementation details I understand that it belongs to trigger element params. Otherwise, I would expect the tooltip itself to be disabled when `disabled: true` is passed to `useTooltip`

Reply:
Yeah that's a good point too. The obvious alternatives aren't clicking for me as good APIs. Let me think about it a bit, but if you have any other ideas I'm open. My main thing is, if we use the DOM as the source of truth we still need it to be reactive, so the effect would probably need a mutation observer or something.

Reply:
The more I think about this, the less concerned I am about the distinction between a disabled trigger and a disabled tooltip. Here's why:

There really isn't a concept of a disabled tooltip IMO. [W3C describes the `aria-disabled` state](https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-disabled) as an "element [that] is perceivable but disabled, so it is not editable or otherwise operable".

A tooltip is never really operable. It's not a control, but meant to provide supplementary content that is never supposed to be interactive anyway. Presumably if it were disabled, it wouldn't pop up at all which means its content would no longer be perceivable.

If for whatever reason someone decided that they had a valid use case for turning off a tooltip, they should probably compose their own with `useTooltip` and render it conditionally. Using `disabled` to describe a tooltip seems like it would always be semantically wrong.

Lastly, the other props accepted by `useTooltip` are composed by the trigger. Props intended for the tooltip itself can be passed directly to `TooltipPopup`. So to me it makes sense that `disabled` would be passed to the trigger in this case.